hibernate-core/src/test/java/org/hibernate/orm/test/query/hql/JpaComplianceDisallowFQNTests.java

```diff
@@ -0,0 +1,43 @@
+package org.hibernate.orm.test.query.hql;
+
+import org.hibernate.orm.test.query.sqm.BaseSqmUnitTest;
+import org.hibernate.orm.test.query.sqm.domain.Person;
+import org.hibernate.query.sqm.StrictJpaComplianceViolation;
+import org.hibernate.testing.orm.junit.ExpectedException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Nahtan Xu
+ */
+@SuppressWarnings("WeakerAccess")
+public class JpaComplianceDisallowFQNTests extends BaseSqmUnitTest {
+
+	@Override
+	protected boolean strictJpaCompliance() {
+		return true;
+	}
+
+	@Override
+	protected Class[] getAnnotatedClasses() {
+		return new Class[] { Person.class };
+	}
+
+	@Test
+	@ExpectedException(StrictJpaComplianceViolation.class)
+	public void testQuery() {
+		interpretSelect( String.format( "select p from %s p", Person.class.getName() ) );
+	}
+	
+	@Test
+	@ExpectedException(StrictJpaComplianceViolation.class)
+	public void testUpdate() {
+		interpretSelect( String.format( "update %s set numberOfToes = 0", Person.class.getName() ) );
+	}
+	
+	@Test
+	@ExpectedException(StrictJpaComplianceViolation.class)
+	public void testDelete() {
+		interpretSelect( String.format( "delete %s", Person.class.getName() ) );
+	}
+	
+}
```

Comment:
A little bit surprised `@Nested` is not allowed (seems `@Test` is the only expected annotation on top level)